File: src/lecture04/java/Main.java
package lecture04.java;

public class Main {

    public static void main(String[] args) {
        JavaMoney money1 = new JavaMoney(2_000L);
        JavaMoney money2 = new JavaMoney(1_000L);
        JavaMoney money3 = new JavaMoney(1_000L);
        JavaMoney money4 = money2;

        System.out.println(money1.compareTo(money2) > 0);
        System.out.println("money2 == money3 = " + (money2 == money3));
        System.out.println("money2 == money4 = " + (money2 == money4));
        System.out.println("money2 equals money3 = " + money2.equals(money3));
        System.out.println("money2 equals money4 = " + money2.equals(money4));

        System.out.println(money1.plus(money2));
    }
}

File: src/lecture04/kotlin/Main.kt
package lecture04.kotlin

import lecture04.java.JavaMoney

fun fun1(): Boolean {
    println("fun1")
    return true
}

fun fun2(): Boolean {
    println("fun2")
    return true
}

fun main() {
    val money1 = JavaMoney(2_000L)
    val money2 = JavaMoney(1_000L)
    val money3 = JavaMoney(1_000L)
    val money4 = money2

    println(money1 > money2)
    println(money2 === money3)
    println(money2 === money4)
    println(money2 == money3)
    println(money2 == money4)

    if (fun1() || fun2()) {
        println("hello")
    }

    println(money1 + money2)
}
